src/main/java/org/dataloader/ValueCache.java

```diff
@@ -140,4 +147,15 @@
      * @return a void future for error handling and fluent composition
      */
     CompletableFuture<Void> clear();
+
+
+    /**
+     * This special exception can be used to short-circuit a caching method
+     */
+    class ValueCachingNotSupported extends UnsupportedOperationException {
+        @Override
+        public synchronized Throwable fillInStackTrace() {
+            return this;
+        }
```

Comment:
This is magic - the exception has no stack trace (we don't want one) AND its fast to allocate because of this.

Reply:
Does it need it to be synchronized?

Reply:
No - it is in the base class.  I will change this